### PR TITLE
fix-handling-of-set_window_geometry

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Improves experimental xdg-shell support.

Drag resizing of xdg-shell clients works on F27 and Artful (with some flickering)

On bionic weston-terminal seems happy. But there are some remaining issues:

Qt apps hang after resize (pre-existing)
GTK apps crash on interaction (pre-existing)

### CODE REVIEW

File: src/server/frontend/wayland/wl_mir_window.h [hunk 1]
```diff
@@ -20,7 +20,9 @@
 #define MIR_FRONTEND_WL_MIR_WINDOW_H
 
 #include "mir/frontend/surface_id.h"
+#include "mir/geometry/displacement.h"
```
Comment (wmww) [inline src/server/frontend/wayland/wl_mir_window.h]: Nitpick: include not needed.

**Comment by AlanGriffiths** [comment]
bors =wmww

**Comment by Saviq** [comment]
bors=wmww

**Comment by Saviq** [comment]
bors r=wmww

**Pull request merged**